wolvic/java/org/chromium/wolvic/Tab.java

```diff
@@ -172,6 +179,40 @@ private int findBackForwardNavigationOffset(NavigationDirection direction) {
         return offset;
     }
 
+    private static String fixupMimeType(String mimeType) {
+        return TextUtils.isEmpty(mimeType) ? "text/html" : mimeType;
+    }
+
+    private static String fixupData(String data) {
+        return TextUtils.isEmpty(data) ? "" : data;
+    }
+
+    /**
+     * Load url without fixing up the url string. Consumers are responsible for ensuring the URL
+     * passed in is properly formatted (i.e. the scheme has been added if left off during user input).
+     *
+     * @param params Parameters for this load.
+     */
+    private void loadUrl(LoadUrlParams params) {
+	// Based on AwContents::loadUrl
+        if (params.getLoadUrlType() == LoadURLType.DATA && !params.isBaseUrlDataScheme()) {
+            // This allows data URLs with a non-data base URL access to file:///android_asset/ and
+            // file:///android_res/ URLs.
+            params.setCanLoadLocalResources(true);
+        }
+
+        // If we are reloading the same url, then set transition type as reload.
+        if (params.getUrl() != null
+                && params.getUrl().equals(mWebContents.getLastCommittedUrl().getSpec())
+                && params.getTransitionType() == PageTransition.TYPED) {
+            params.setTransitionType(PageTransition.RELOAD);
+        }
+        params.setTransitionType(params.getTransitionType() | PageTransition.FROM_API);
+        params.setUrl(UrlFormatter.fixupUrl(params.getUrl()).getPossiblyInvalidSpec());
```

Comment:
You said in the comments that this method loads the url without fixing up the url string, but here yu are updating the param's url.

Reply:
Makes sense, I'll modify the comment accordingly

Reply:
Finally removed it since it's a private method after all.